aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteModelFieldTypeConverter.java

```diff
@@ -0,0 +1,242 @@
+        for (Map.Entry<String, ModelField> entry : innerModelSchema.getFields().entrySet()) {
+            mapForInnerModel.put(entry.getKey(), nestedModelConverter.convertValueFromSource(cursor, entry.getValue()));
+        }
+        final String modelInJsonFormat = gson.toJson(mapForInnerModel);
+        try {
+            return gson.getAdapter(nestedModelType).fromJson(modelInJsonFormat);
+        } catch (IOException exception) {
+            LOGGER.warn("", exception);
+            return null;
+        }
+    }
+
+    private Object convertCustomTypeToTarget(Cursor cursor, ModelField field, int columnIndex) throws IOException {
+        final String stringValue = cursor.getString(columnIndex);
+        return gson.getAdapter(field.getType()).fromJson(stringValue);
+    }
+
+    private Object convertDateToTarget(Cursor cursor, ModelField field, int columnIndex) throws ParseException {
+        // TODO wire up the new Date/Time handling here
+        final String dateInStringFormat = cursor.getString(columnIndex);
+        if (dateInStringFormat != null) {
+            return SimpleDateFormat
+                    .getDateInstance()
+                    .parse(dateInStringFormat);
+        }
+        return null;
+    }
+
+    @Override
+    public Object convertValueFromTarget(Model model, ModelField field) throws DataStoreException {
+        final String fieldName = field.getName();
+
+        final Object fieldValue = getModelValue(model, fieldName);
+        if (fieldValue == null) {
+            return null;
+        }
+        final JavaFieldType javaFieldType = TypeConverter.getJavaFieldType(field);
+
+        switch (javaFieldType) {
+            case INTEGER:
+            case LONG:
+            case FLOAT:
+            case STRING:
+                // these types require no special treatment
+                return fieldValue;
+            case BOOLEAN:
+                boolean booleanValue = (boolean) fieldValue;
+                return booleanValue ? 1L : 0L;
+            case MODEL:
+                return ((Model) fieldValue).getId();
+            case ENUM:
+            case CUSTOM_TYPE:
+                return gson.toJson(fieldValue);
+            case DATE:
+                // TODO integrate with new Date/Time handling
+                final Date dateValue = (Date) fieldValue;
+                return SimpleDateFormat
+                        .getDateInstance()
+                        .format(dateValue);
+            case TIME:
+                return ((Time) fieldValue).getTime();
+            default:
+                LOGGER.warn("Field of type " + javaFieldType + " is not supported. Fallback to null.");
+                return null;
+        }
+    }
+
+    private Object getModelValue(Model model, String fieldName) throws DataStoreException {
+        Class<? extends Model> modelClass = model.getClass();
+        final String getterName = "get" + Character.toUpperCase(fieldName.charAt(0)) + fieldName.substring(1);
```

Comment:
This is sort of a rule about the model classes. I wonder if we could make this a utility on one of them. Like `ModelSchema.getterForField(fieldName)`?  Maybe not `ModelSchema` itself...

Reply:
I created a `ModelHelper`... I'm not a big fan of "helper", "utils" stuff in general, but sometimes I just find myself with a better alternative, like in this case. So, feel free to suggest something different :)

Reply:
Ah gosh actually we have this [`FieldFinder`](https://github.com/aws-amplify/amplify-android/blob/master/core/src/main/java/com/amplifyframework/util/FieldFinder.java#L72) already too, that I'd forgotten about. It extracts a value from an object, by looking up an instance variable by name.

Maybe let's punt this for now. We could do a separate, targeted PR that lumps together some of the reflection utilities that are currently around the code-base.

Reply:
Agree. I'll merge my code as is and follow up with a clean-up/consolidation of all the reflection related code DataStore has